lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java

```diff
@@ -69,8 +82,10 @@ public static String defaultVersion() {
 		private final boolean isScript;
 		/** The jar that contains the eclipse formatter. */
 		final JarState jarState;
+		private final Map<String, String> userData;
 
-		State(String version, Provisioner provisioner, boolean isScript) throws IOException {
+		State(String version, Provisioner provisioner, boolean isScript, Map<String, String> userData) throws IOException {
```

Comment:
When you store the `Map<String, String>`, I would copy it into a `SortedMap<String, String>` first.  Up-to-date checking is not based on `.equals()`, but rather on binary-equals of the serialized representation.  If you make a `Map<String, String>`, the ordering might change depending on the bucket size, even though the data hasn't changed.